src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.exeptions.NonAnnotationException;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NonAnnotationException {
+        Constructor declaredConstructor = clazz.getDeclaredConstructor();
+        Object newInstance = declaredConstructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                if (field.getType().equals(BetDao.class)
+                        && BetDaoImpl.class.isAnnotationPresent(Dao.class)) {
+                    field.setAccessible(true);
+                    field.set(newInstance, Factory.getBetDao());
+                } else if (field.getType().equals(UserDao.class)
+                        && UserDaoImpl.class.isAnnotationPresent(Dao.class)) {
+                    field.setAccessible(true);
+                    field.set(newInstance, Factory.getUserDao());
+                } else {
+                    throw new NonAnnotationException("Відсутня антотація @Dao у класі.");
+                }
```

Comment:
looks good
I would only change one thing:

```suggestion
                if (field.getType().equals(BetDao.class)
                        && Factory.getBetDao().getClass().isAnnotationPresent(Dao.class)) {
                    field.setAccessible(true);
                    field.set(newInstance, Factory.getBetDao());
                } else if (field.getType().equals(UserDao.class)
                        && Factory.getUserDao().getClass().isAnnotationPresent(Dao.class)) {
                    field.setAccessible(true);
                    field.set(newInstance, Factory.getUserDao());
                } else {
                    throw new NonAnnotationException("Відсутня антотація @Dao у класі.");
                }
```

Reply:
done